fs_src/index.html

```diff
@@ -1,595 +1,551 @@
-          <label><span id="toggle_spinner"></span>Toggle</label>
+      `;
+    }
+
+    function InputBlock({label, value, type = "text", placeholder = null, accept = null, onInput = null, pref = null}) {
+      return html`
+        <${FormBlock} label="${label}">
+          <input type="${type}" value="${value}" placeholder="${placeholder}" accept="${accept}" onInput="${onInput}" ref="${pref}"/>
+        <//>
+      `;
+    }
+
+    function SwitchBlock({label, checked, spin, onClick = null, children}) {
+      return html`
+        <${FormBlock} label="${label}">
+          <label class="switch">
+            <input type="checkbox" checked="${checked}" onClick=${onClick}/>
+            <span class="slider round"></span>
+          </label>
+          <div class="spinner_container">
+            <span class="spinner ${spin && 'spin'}"></span>
+          </div>
+          ${children}
+        <//>
+      `;
+    }
+
+    function SelectBlock({label, select, options, onChange}) {
+      return html`
+        <${FormBlock} label="${label}">
+          <select onChange="${onChange}">
+            ${Object.entries(options).map(([value, label]) => html`
+              <option value="${value}" ...${{selected: (select == value)}}>${label}</option>
+            `)}
+          </select>
+        <//>
+      `;
+    }
+
+    function InfoBlock({label, info}) {
+      return html`
+        <${FormBlock} label="${label}">
+          <span>${info}</span>
+        <//>
+      `;
+    }
+
+    function Button({label, spin, onClick}) {
+      return html`
+        <button class="btn" onClick=${onClick}>
+          <label><span class="spinner ${spin && 'spin'}"></span>${label}</label>
         </button>
-      </div>
-    </div>
-  </div>
-</div>
-<!-- script.js inserted below during build -->
-<script src="script.js"></script>
+      `;
+    }
+
+    function ButtonBlock(props) {
+      return html`
+        <${FormBlock}>
+          <${Button} ...${props}/>
+        </div>
+      `;
+    }
+
+    class ApiClient {
+      requestId = 0
+      callbacks = {}
+
+      bind(app) {
+        this.app = app;
+      }
+
+      connect() {
+        this.socket = new WebSocket(`ws://${this.app.state.remoteHost}/rpc`);
+        this.socket.onopen = () => {
+          this.getInfo();
+          setInterval(() => this.getInfo(), 1000);
+        }
+        this.socket.onerror = () => this.updateState({connected: false});
+        this.socket.onclose = () => {
+          this.updateState({connected: false});
+          setInterval(() => this.reconnect(), 1000);
+        };
+        this.socket.addEventListener('message', ((responce) => this.onMessage(responce)));
+      }
+
+      reconnect() {
+        this.socket = new WebSocket(`ws://${this.app.state.remoteHost}/rpc`);
+        this.socket.onerror = () => {}
+        this.socket.onopen = () => location.reload();
+      }
+
+      send(method, params = {}, callback = null) {
+        let requestId = this.requestId++;

```

Comment:
id 0 is not a valid request id, should start at 1 (or better yet, a random integer).